tests/vision/checks/performance/single_dataset_scalar_performance_test.py

```diff
@@ -0,0 +1,92 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+
+import warnings
+
+import torch
+from hamcrest import (assert_that, calling, equal_to, greater_than_or_equal_to,
+                      raises)
+from ignite.metrics import Accuracy, Precision
+
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.vision.checks.performance.single_dataset_scalar_performance import \
+    SingleDatasetScalarPerformance
+
+
+def test_detection_deafults(coco_train_visiondata, mock_trained_yolov5_object_detection, device):
+    # Arrange
+    check = SingleDatasetScalarPerformance()
+
+    # Act
+    result = check.run(coco_train_visiondata, mock_trained_yolov5_object_detection, device=device)
+
+    # Assert
+    # scalar
+    assert_that(type(result.value['score']), equal_to(float))
+    # metric
+    assert_that(result.value['score'], greater_than_or_equal_to(0))
+
+
+def test_detection_w_params(coco_train_visiondata, mock_trained_yolov5_object_detection, device):
+    # params that should run normally
+    check = SingleDatasetScalarPerformance(reduce=torch.max, reduce_name='torch_max')
+    result = check.run(coco_train_visiondata, mock_trained_yolov5_object_detection, device=device)
+    assert_that(type(result.value['score']), equal_to(float))
+
+
+def test_classification_defaults(mnist_dataset_train, mock_trained_mnist, device):
+    # Arrange
+    check = SingleDatasetScalarPerformance()
+
+    # Act
+    result = check.run(mnist_dataset_train, mock_trained_mnist, device=device)
+
+    # Assert
+    # scalar
+    assert_that(type(result.value['score']), equal_to(float))
+    # metric
+    assert_that(result.value['score'], greater_than_or_equal_to(0))
+
+
+def test_add_condition(mnist_dataset_train, mock_trained_mnist, device):
+    # Arrange
+    check = SingleDatasetScalarPerformance()
+    check.add_condition_greater_than(0.5)
+    check.add_condition_less_equal_to(0.2)
+
+    # Act
+    result = check.run(mnist_dataset_train, mock_trained_mnist, device=device)
+
+    # Assert
+    assert_that(result.conditions_results[0].is_pass)
```

Comment:
also we try to use less test function for vision so you can add the condition to the same function that asserts value and assert the condition results there